src/site/stages/build/drupal/api.js

```diff
@@ -37,10 +38,22 @@ function getDrupalClient(buildOptions) {
   const agent = new SocksProxyAgent('socks://127.0.0.1:2001');
 
   return {
-    // We have to point to aws urls on Jenkins, so the only
-    // time we'll be using cms.va.gov addresses is locally,
-    // when we need a proxy
-    usingProxy: address.includes('cms.va.gov'),
+    // During Production WEB builds, the WEB build process reads from the Drupal
+    // CMS using the AWS ELB URL.
+    // e.g. internal-dsva-vagov-prod-cms-2000800896.us-gov-west-1.elb.amazonaws.com
+
+    // When running WEB build process, the site at DRUPAL_ADDRESS must be reachable.
+    // If building and sourcing content from a site at *.cms.va.gov, from outside of VAEC,
+    // the socks proxy agent must be used because those sites are in VAEC.
+
+    // If a CMS is available at DRUPAL_ADDRESS, reachable by the WEB build process, the proxy is not needed.
+    // This includes in local development environment where user has a local CMS instance
+    // and when WEB is built in CMS-CI where every WEB has a CMS instance.
```

Comment:
This is a lot to digest. While I appreciate the level of detail, I'm not sure I understand it all.

My main concern is usability; if I tell the build script to `--use-proxy` when I shouldn't or don't `--use-proxy` when I should, is it going to error out? If so, does that error make it clear what I should do to fix it?

If it's user-friendly, then having this level of detail in here makes enough sense. If it isn't, I'll have to hunt through the code to figure out what the story is, and when I run into this, it may not be clear to me:
1) That this comment block is related to my problem
2) What I should do to fix it

Reply:
I think the confusion for us comes from the fact that devops/infrastructure/application boundaries are vague here.

This proxy is setup using the application code, and the URLs are hard coded for the environment.

In our world, we design the app to assume the connection is available, and setup the infrastructure (or local development tools) to setup the proxy if needed.

Reply:
@cvalarida Big picture aside, do you like the useProxy option? We can set whatever default you want it to have. We need it for CI so having a CLI option is fine for us.

If I update the comment here to be more descriptive, would the useProxy parts be mergable?

Reply:
Ok, @cvalarida I tested out WEB master branch and we do need to figure out the Proxy defaults.

In our CI system, it defaults to enabled. See the web build failure here:
> FetchError: request to http://pr595.ci.cms.va.gov/flags_list failed, reason: connect ECONNREFUSED 127.0.0.1:2001


https://github.com/department-of-veterans-affairs/va.gov-cms/pull/595#commitcomment-35199205

Reply:
I added a `--no-drupal-proxy` flag a couple days ago. Does that solve this problem?